hubblestack/daemon.py

```diff
@@ -214,6 +231,22 @@ def schedule():
             log.error('Scheduled job {0} has an invalid value for seconds or '
                       'splay.'.format(jobname))
         args = jobdata.get('args', [])
+        if 'randomizeSplay' in jobdata and jobdata['randomizeSplay']:
+           if 'buckets' not in jobdata or int(jobdata['buckets'])<=0:
+              buckets = int(256)
+           else:
+              buckets = int(jobdata['buckets'])
+           try:
+              hostname = socket.gethostbyname(socket.gethostname())
+              ipValues = hostname.split('.')
+              ipValueSum = int(ipValues[0])+int(ipValues[1])+int(ipValues[2])+int(ipValues[3])
```

Comment:
I don't know if summing the ip addresses will actually result in a normal distribution of servers. Why not just use the random library to let it decide the normal distribution?

```
bucketnumber = random.randint(1, buckets + 1)
```

Reply:
The intent here is to place the machines evenly in all buckets. A random bucket number could lead to a situation where a lot of machines are put in a single bucket and a lot of buckets are empty. 

The IP was chosen here because usually in an industry VMs have consecutive IPs(not guaranteed though). This would help in the normal distribution of machines in the buckets.